### ISSUE TITLE
Magisk installer timeouts pointless

### ISSUE THREAD

**Issue by DoganM95** [Magisk-Modules-Repo/HideNavBar#61]
The magisk module installer asks questions like "choose gesture mode" and if the user does not react within 3 seconds or so, it just goes on to the next step/fails the installation. 

Instead, it should wait for user input indefinitely, until it proceeds to any next step. 3 seconds are sometimes not enough to read + make a useful decision.

The current installation process feels like a speedrun :)

**Comment by DanGLES3** [comment]
> The magisk module installer asks questions like "choose gesture mode" and if the user does not react within 3 seconds or so, it just goes on to the next step/fails the installation. 
> 
> Instead, it should wait for user input indefinitely, until it proceeds to any next step. 3 seconds are sometimes not enough to read + make a useful decision.
> 
> The current installation process feels like a speedrun :)

The timeout is an intentional choice by the developer of the volume selector as the regular method of detecting key inputs may not be detected so it falls back to an older mode upon the timeout

**Comment by DoganM95** [comment]
Would appreciate it, if you increase it to like a minute on each interaction, so the user has enough time, because i don't see any advantage in a cancellation of the selection on missing user input 😅

thanks!

**Comment by DanGLES3** [comment]
> Would appreciate it, if you increase it to like a minute on each interaction, so the user has enough time, because i don't see any advantage in a cancellation of the selection on missing user input 😅
> 
> thanks!

Point being from my knowledge it times out so the alternative volume handler kicks in for devices which do not support the current one

Wouldn't be very great if some users had to wait 1 minute before being able to select anything

**Comment by smemsh** [comment]
I agree that the timeout makes me feel like I'm in a race, and doesn't give enough time for new users to consider options they've never seen before.  I had to try three times and by the end actually had to memorize my presses ahead of time so I could press them quickly enough in succession to beat the timeouts.

One idea is to use a dummy prompt with some initial question, like "ready to proceed?" and then save the determined volume-press method so it can be applied to all later questions without them needing any timeout themselves.

**Comment by DanGLES3** [comment]
> I agree that the timeout makes me feel like I'm in a race, and doesn't give enough time for new users to consider options they've never seen before. I had to try three times and by the end actually had to memorize my presses ahead of time so I could press them quickly enough in succession to beat the timeouts.
> 
> One idea is to use a dummy prompt with some initial question, like "ready to proceed?" and then save the determined volume-press method so it can be applied to all later questions without them needing any timeout themselves.

That would require directly modding the volume selector tool script, also that first question would still need a timeout (so it can fallback to the alternative check method)

**Comment by smemsh** [comment]
Actually, if `timeout` was not used, and -- for whatever reason -- neither `getevent` nor `keycheck` worked properly, it could hang.  So, your original idea to just increase the timeout is probably best.  I think 10 seconds would be good enough, personally.

**Comment by DanGLES3** [comment]
> Actually, if `timeout` was not used, and -- for whatever reason -- neither `getevent` nor `keycheck` worked properly, it could hang. So, your original idea to just increase the timeout is probably best. I think 10 seconds would be good enough, personally.

It was technically already set to 10 not sure why it's lower for some people, so I increased it to 15

**Comment by jacoghi** [comment]
This timeout is really bad, in my phone I barely have enough time to read the sentence and it's already timing out, not even 3 seconds. Maybe adding to the Readme the order the questions will show up so users can come up with their own street fighter cheat code to get the configuration right.

**Comment by DanGLES3** [comment]
> This timeout is really bad, in my phone I barely have enough time to read the sentence and it's already timing out, not even 3 seconds. Maybe adding to the Readme the order the questions will show up so users can come up with their own street fighter cheat code to get the configuration right.

That wouldn't solve anything, and tbh would bloat the readme even further, i can only hope the volume selector gets fixed

**Comment by jacoghi** [comment]
It definitely doesn't solve anything, but the problem is that without a cheat code, I literally had to go over the installation 5 times. Twice to understand that there was a timeout. 3rd was to try to get it done in a rush, which I failed miserably, then 4th time I picked anything just se I could at least see the questions, otherwise I wouldn't even see the questions. 5th time was with the cheat code after writing down a combination during the 4th attempt. Now that it's installed, works beautifully (and really thanks for this, I remember always adding that line to the system.prop file to achieve the same result prior to the read only system images), but the installation process leaves a lot to be desired the way it's currently set up. Thanks for this, great module.

**Comment by smemsh** [comment]
I had the same experience.  I have to keep a document with the list of responses and enter them quickly, every time I use it.  I think it's not waiting 10 seconds (now 15), but I stopped chasing it.

definitely invaluable module though... really amazes me google still has persistent pill bar after all these years, burning a line in everyone's oled screen, with no way to disable it
